packages/react/src/components/Comments/ThreadsSidebar.tsx

```diff
@@ -117,66 +180,47 @@ export function ThreadsSidebar(props: {
       state?.threadPositions
     );
 
-    const openThreads: ThreadData[] = [];
-    const resolvedThreads: ThreadData[] = [];
+    const ret: Array<{ thread: ThreadData; referenceText: string }> = [];
```

Comment:
How come we store the `referenceText` here instead of just calling `getReferenceText` when we render the `ThreadItem`s? Ig it's about preventing unnecessary rerenders?